src/Data/NodeResolver.php

```diff
@@ -582,6 +588,30 @@ public function parse_request( string $uri, $extra_query_vars = '' ) {
 		return $uri;
 	}
 
+	/**
+	 * Given a WP Post or post ID, this method attempts to resolve a preview post ID.
+	 * 
+	 * @param Int|WP_Post $post The WP Post object or Post ID
+	 * 
+	 * @return Int A preview post ID if one exists, the current post ID if one doesn't exist.
+	 */
+	public static function get_post_preview_id( $post ) {
```

Comment:
Now that this is a public static getter, I think we should move this to `WPGraphQL\Utils`.

It's not directly related to NodeResolver, it's the only static method on the class, and as a public static, keeping it with other utils methods will increase refactorability.